### PR TITLE
Migrate "arbitrary_trial_specific_annotations" to "clinical"

### PR DESCRIPTION
**Pull request by scvannost**

### CODE REVIEW

File: cidc_schemas/schemas/participant.json [hunk 1]
```diff
@@ -68,7 +68,7 @@
         "Other"
       ]
     },
-    "arbitrary_trial_specific_clinical_annotations":{
+    "trial_specific":{
```
Comment (jacoblurye) [inline cidc_schemas/schemas/participant.json:71]: My preference is for renaming to `clinical` over `trial_specific`.

**Review by jacoblurye — APPROVED** [review]
Looks good!

File: cidc_schemas/schemas/participant.json (hunk 1 above)
Comment (scvannost) [reply cidc_schemas/schemas/participant.json]: fixed in next commit

File: docs/docs/participant.html [hunk 2]
```diff
@@ -424,8 +424,8 @@ <h3>Properties</h3>
                 <tr>
     <td>
         
-            <span id="arbitrary_trial_specific_clinical_annotations">
-                arbitrary_trial_specific_clinical_annotations
+            <span id="trial_specifc">
+                trial_specifc
```
Comment (scvannost) [inline docs/docs/participant.html:428]:
```suggestion
            <span id="clinical">
                clinical
```

**Review by roshni-b — APPROVED** [review]

**Pull request merged**